service/main.go

```diff
@@ -52,13 +53,14 @@ func getConfig() Config {
 func main() {
 	config := getConfig()
 	store := s3.New(config.Bucket, config.Profile)
+	codeStore := s3.New(config.CodeBucket, config.Profile)
 
 	http.HandleFunc("/healthcheck", middleware.WithRequestLog(http.HandlerFunc(ok)))
 
 	if config.RequireAuth {
-		http.Handle("/", middleware.WithRequestLog(middleware.WithAuth(middleware.WithDomainPrefix(storeServer(store)))))
+		http.Handle("/", middleware.WithRequestLog(middleware.WithAuth(middleware.WithDomainPrefix(storeServer(store, codeStore)))))
```

Comment:
This is the bit that makes me somewhat uneasy: we are potentially serving the prod bucket via a code domain. The prefixe match should prevent this from happening but fundamentally the CODE path is still attempting to hit PROD data and this doesn't quite sit right.